services/pagerduty2/service.go

```diff
@@ -0,0 +1,328 @@
+}
+
+// Global is a bound method of the Service struct, returns whether the Service configuration is global
+func (s *Service) Global() bool {
+	c := s.config()
+	return c.Global
+}
+
+type testOptions struct {
+	IncidentKey string          `json:"incident-key"`
+	Description string          `json:"description"`
+	Details     string          `json:"details"`
+	Level       alert.Level     `json:"level"`
+	Data        alert.EventData `json:"event_data"`
+	Timestamp   time.Time       `json:"timestamp"`
+}
+
+// TestOptions returns optional values for the test harness
+func (s *Service) TestOptions() interface{} {
+	layout := "2006-01-02T15:04:05.000Z"
+	str := "2014-11-12T11:45:26.371Z"
+	t, _ := time.Parse(layout, str)
+	return &testOptions{
+		IncidentKey: "testIncidentKey",
+		Description: "test pagerduty2 message",
+		Level:       alert.Critical,
+		Timestamp:   t,
+		Data: alert.EventData{
+			Name:   "testPagerDuty2",
+			Tags:   make(map[string]string),
+			Fields: make(map[string]interface{}),
+			Result: models.Result{},
+		},
+	}
+}
+
+// Test is a bound method of the Service struct that handles testing the Alert function
+func (s *Service) Test(options interface{}) error {
+	o, ok := options.(*testOptions)
+	if !ok {
+		return fmt.Errorf("unexpected options type %T", options)
+	}
+	c := s.config()
+	return s.Alert(
+		c.ServiceKey,
+		o.IncidentKey,
+		o.Description,
+		o.Level,
+		o.Details,
+		o.Timestamp,
+		o.Data,
+	)
+}
+
+// Alert is a bound method of the Service struct that processes a given alert to PagerDuty
+func (s *Service) Alert(serviceKey, incidentKey, desc string, level alert.Level, details string, timestamp time.Time, data alert.EventData) error {
+	url, post, err := s.preparePost(serviceKey, incidentKey, desc, level, details, timestamp, data)
+	if err != nil {
+		return err
+	}
+
+	resp, err := http.Post(url, "application/json", post)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusOK {
+		body, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return err
+		}
+		type response struct {
+			Message string `json:"message"`
+		}
+		r := &response{Message: fmt.Sprintf("failed to understand PagerDuty2 response. code: %d content: %s", resp.StatusCode, string(body))}
+		b := bytes.NewReader(body)
+		dec := json.NewDecoder(b)
+		dec.Decode(r)
+		return errors.New(r.Message)
+	}
+	return nil
+}
+
+// preparePost is a helper method that sets up the payload for transmission to PagerDuty
+func (s *Service) preparePost(serviceKey, incidentKey, desc string, level alert.Level, details string, timestamp time.Time, data alert.EventData) (string, io.Reader, error) {
+	c := s.config()
+	if !c.Enabled {
+		return "", nil, errors.New("service is not enabled")
+	}
+
+	var severity string
+	eventType := "trigger"
+
+	switch level {
+	case alert.Warning:
+		severity = "warning"
+	case alert.Critical:
+		severity = "critical"
+	case alert.Info:
+		severity = "info"
+	case alert.Error:
```

Comment:
How does the alert.Error level get set?

Reply:
Good catch, this is my reading of the API v2 docs and not the internal representation from the `kapacitor` side...I'll remove.